Sources/Authgear.swift

```diff
@@ -285,6 +285,7 @@ public class Authgear {
 
     private let jwkStore = JWKStore()
     private let workerQueue: DispatchQueue
+    private let accessTokenLock = NSLock()
```

Comment:
Why do we need this?

Reply:
This is only used for the syncing when changing the access token, as shown in 71b0d94

Reply:
- We still need `DispatchQueue.main.async` to ensure the changes is made on main thread.
- `DispatchQueue.main.async` should already ensure the changes are made sequentially. Did you observe any race condition that would require explicit synchronization?

Reply:
No, I noticed explicit synchronization was used for the [android sdk](https://github.com/authgear/authgear-sdk-android/blob/33e2e38851edc62620a81c90b549099a75cd5b71/sdk/src/main/java/com/oursky/authgear/AuthgearCore.kt#L594-L609), so I follow the same pattern here. If  this is not necessary, I can remove this change

Reply:
If you observed no problem with just `DispatchQueue.main.async`, can remove the lock.

Reply:
Lock removed